### PR TITLE
Feature/storage power actor 2

### PR DESCRIPTION
**Pull request by Alexey-N-Chernyshov**
### Description of the Change
Storage Power Actor methods 2-8 implemented and tested.

### Possible Drawbacks 
More PR with methods upcoming.

### CODE REVIEW

File: core/vm/actor/builtin/storage_power/storage_power_actor_export.cpp [hunk 1]
```diff
@@ -99,10 +101,175 @@ ACTOR_METHOD(StoragePowerActorMethods::withdrawBalance) {
...
+      create_miner_params.worker,
+      create_miner_params.sector_size,
+      create_miner_params.peer_id};
+  OUTCOME_TRY(encoded_construct_miner_parameters,
+              encodeActorParams(construct_miner_parameters));
+  init::ExecParams exec_parameters{kStorageMinerCodeCid,
+                                   encoded_construct_miner_parameters};
+  OUTCOME_TRY(encoded_exec_parameters, encodeActorParams(exec_parameters));
+  OUTCOME_TRY(encoded_addresses_created,
+              runtime.send(kInitAddress,
```
Comment (turuslan) [inline core/vm/actor/builtin/storage_power/storage_power_actor_export.cpp]:
```
runtime.sendPR<init::ExecReturn>(
  kInitAddress,
  init::kExecMethodNumber,
  init::ExecParams{
    .code = kStorageMinerCodeCid,
    .params = encoded_construct_miner_parameters,
  },
  0)
```

File: core/vm/actor/builtin/storage_power/storage_power_actor_export.cpp [hunk 2]
```diff
@@ -99,10 +101,183 @@ ACTOR_METHOD(StoragePowerActorMethods::withdrawBalance) {
...
 
-  // commit state
   OUTCOME_TRY(power_actor_state, power_actor.flushState());
-  OUTCOME_TRY(state_cid, datastore->setCbor(power_actor_state));
-  OUTCOME_TRY(runtime.commit(ActorSubstateCID{state_cid}));
+  OUTCOME_TRY(runtime.commitState(power_actor_state));
+  return fc::outcome::success();
+}
+
+fc::outcome::result<InvocationOutput> StoragePowerActorMethods::createMiner(
```
Comment (turuslan) [inline core/vm/actor/builtin/storage_power/storage_power_actor_export.cpp]: `ACTOR_METHOD(StoragePowerActorMethods::createMiner) {`

File: core/vm/actor/builtin/storage_power/storage_power_actor_export.cpp [hunk 3]
```diff
@@ -99,10 +101,183 @@ ACTOR_METHOD(StoragePowerActorMethods::withdrawBalance) {
...
+  OUTCOME_TRY(encoded_addresses_created,
+              runtime.send(kInitAddress,
+                           init::kExecMethodNumber,
+                           encoded_exec_parameters,
+                           TokenAmount{0}));
+  OUTCOME_TRY(addresses_created,
+              decodeActorReturn<init::ExecReturn>(encoded_addresses_created));
+
+  auto datastore = runtime.getIpfsDatastore();
+  OUTCOME_TRY(state, datastore->getCbor<StoragePowerActorState>(actor.head));
```
Comment (turuslan) [inline core/vm/actor/builtin/storage_power/storage_power_actor_export.cpp]: `getCurrentActorStateCbor`

File: core/vm/actor/builtin/storage_power/storage_power_actor_export.cpp (hunk 1 above)
Comment (Alexey-N-Chernyshov) [reply core/vm/actor/builtin/storage_power/storage_power_actor_export.cpp]: done

File: core/vm/actor/builtin/storage_power/storage_power_actor_export.cpp (hunk 2 above)
Comment (Alexey-N-Chernyshov) [reply core/vm/actor/builtin/storage_power/storage_power_actor_export.cpp]: done

File: core/vm/actor/builtin/storage_power/storage_power_actor_export.cpp (hunk 3 above)
Comment (Alexey-N-Chernyshov) [reply core/vm/actor/builtin/storage_power/storage_power_actor_export.cpp]: done

File: core/vm/runtime/runtime.hpp [hunk 4]
```diff
@@ -166,7 +167,7 @@ namespace fc::vm::runtime {
     /// Send funds
     inline auto sendFunds(const Address &to, BigInt value) {
       // kSendMethodNumber circular dependency
```
Comment (turuslan) [inline core/vm/runtime/runtime.hpp]: remove

File: core/vm/actor/builtin/storage_power/storage_power_actor_export.cpp [hunk 5]
```diff
@@ -99,10 +98,169 @@ ACTOR_METHOD(StoragePowerActorMethods::withdrawBalance) {
...
+
+  StoragePower power =
+      consensusPowerForWeight(on_sector_prove_commit_params.weight);
+  OUTCOME_TRY(network_power, power_actor.getTotalNetworkPower());
+  TokenAmount pledge =
+      pledgeForWeight(on_sector_prove_commit_params.weight, network_power);
+  OUTCOME_TRY(
+      power_actor.addToClaim(runtime.getMessage().get().from, power, pledge));
+
+  OnSectorProveCommitReturn exec_return{pledge};
```
Comment (turuslan) [inline core/vm/actor/builtin/storage_power/storage_power_actor_export.cpp]: rename `exec_return`, or `encodeActorReturn(OnSectorProveCommitReturn{.pledge=pledge})`

File: test/core/vm/actor/builtin/storage_power/storage_power_actor_test.cpp [hunk 6]
```diff
@@ -329,12 +399,368 @@ TEST_F(StoragePowerActorTest, WithdrawBalanceSuccess) {
...
+                      datastore->getCbor<StoragePowerActorState>(state_cid));
+  StoragePowerActor actor(datastore, state);
+  EXPECT_OUTCOME_TRUE(claim, actor.getClaim(miner_address));
+  EXPECT_EQ(claim.pledge, pledge);
+  EXPECT_EQ(claim.power, sector_size);
+}
+
+/**
+ * @givenRuntime and state with miner
+ * @when onSectorProofCommitSuccess called
```
Comment (turuslan) [inline test/core/vm/actor/builtin/storage_power/storage_power_actor_test.cpp]: `onSectorTerminate`

Comment (Alexey-N-Chernyshov) [reply test/core/vm/actor/builtin/storage_power/storage_power_actor_test.cpp]: fixed

File: core/vm/runtime/runtime.hpp (hunk 4 above)
Comment (Alexey-N-Chernyshov) [reply core/vm/runtime/runtime.hpp]: ok

File: core/vm/actor/builtin/storage_power/storage_power_actor_export.cpp (hunk 5 above)
Comment (Alexey-N-Chernyshov) [reply core/vm/actor/builtin/storage_power/storage_power_actor_export.cpp]: ok

File: core/vm/actor/builtin/storage_power/policy.cpp [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+  }
+
+  StoragePower consensusPowerForWeight(const SectorStorageWeightDesc &weight) {
+    return StoragePower{weight.sector_size};
+  }
+
+  StoragePower consensusPowerForWeights(
+      gsl::span<SectorStorageWeightDesc> weights) {
+    StoragePower power{0};
+    for (auto weight : weights) {
```
[the remark this answers is not in the document]
Comment (Alexey-N-Chernyshov) [reply core/vm/actor/builtin/storage_power/policy.cpp]: fixed

**Pull request merged**